internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -156,6 +162,21 @@ func (c *controller) validateConfig() error {
 	return utilerrors.NewAggregate(errs)
 }
 
+// Preflight is a check that should be run before making any changes to the cluster
+type Preflight interface {
+	// Install runs checks that should be successful prior
+	// to installing the Helm release. It is provided
+	// a Helm release and returns an error if the
+	// check is unsuccessful
+	Install(context.Context, *release.Release) error
```

Comment:
Having preflight here looks good, and not blocking this PR at all. I remember this being mentioned somewhere earlier - but just to refresh, can this be directly integrated as a part of helm-operator-plugin's `ActionInterface`: https://github.com/operator-framework/helm-operator-plugins/blob/ae087c419483d73c37155c02f142ab14b4d8c706/pkg/client/actionclient.go#L55?

There may be tweaks required to get the existing release from the cluster and passing it to action client, but this thought popped up because we *may* probably have to expand checks for Uninstall to, and could be cleaner if the helm client can do it all together.

Reply:
I originally had a PR adding it to the actionclient here: https://github.com/operator-framework/helm-operator-plugins/pull/327

There was some discussion in slack where it was decided that it would be a better approach to place the logic directly in operator-controller/rukpak: https://kubernetes.slack.com/archives/C0181L6JYQ2/p1714509603408259

Reply:
Ah Sorry! Now I realize why I remembered it! :see_no_evil: